src/defineCreateClient.ts

```diff
@@ -24,13 +24,13 @@ export default function defineCreateClientExports<
   ClassConstructor: new (httpRequest: HttpRequest, config: ClientConfigType) => SanityClientType,
 ) {
   // Set the http client to use for requests, and its environment specific middleware
-  const defaultRequester = defineHttpRequest(envMiddleware, undefined)
+  const defaultRequester = defineHttpRequest(envMiddleware)
 
   const createClient = (config: ClientConfigType) => {
-    const configRequester = defineHttpRequest(envMiddleware, config)
+    const clientRequester = defineHttpRequest(envMiddleware)
```

Comment:
Is it intentional to not pass config here anymore?

Reply:
Yes, based on the feedback above, I'm opting for a "warn once" pattern. https://github.com/sanity-io/client/pull/925/files#diff-e54beffa7132ee4fc0c37a1b8d3fa97bce3a4aea12db7ec23fdd28c112b3c05bR21-R33

I'm still thinking through how the warning ought to be. To me, it makes sense to limit version warnings to once as they are set on the client. However, we also use the warning header for duplicate parameters which should probably be always printed.

Reply:
But isn't this a breaking change? If you call `createClient(myConfig)`, won't it now entirely ignore `myConfig`?

Reply:
Ah, I guess it's still being applied to the ClassConstructor further down